### PR TITLE
Notifications API Client documentation

### PR DESCRIPTION
**Pull request by agulowaty**
Better view of this change (README):
https://github.com/guardian/mobile-notifications-api-client/tree/MAPI-1064-document-push-api

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -1,3 +1,76 @@
...
+### Create configured Notifications API Client
+
+```scala
+import com.gu.mobile.notifications.client.ApiClient
+
+val httpProvider = <your http provider instance>
+val client = ApiClient(
+  host = "http://notifications-host.com", 
+  apiKey = "API-KEY", 
+  httpProvider,
```
Comment (davidfurey) [inline README.markdown]: If you are going to use named parameters (which you should), best to name all of them

File: README.markdown [hunk 2]
```diff
@@ -1,3 +1,76 @@
...
+
+```scala
+import com.gu.mobile.notifications.client.HttpProvider
+
+object NotificationHttpProvider extends HttpProvider {
+  override def post(url: String, contentType: ContentType, body: Array[Byte]): Future[HttpResponse] = {
+    WS.url(url)
+      .withHeaders("Content-Type" -> s"${contentType.mediaType}; charset=${contentType.charset}")
+      .post(body)
+      .map(extractHttpResponse)
```
Comment (davidfurey) [inline README.markdown]: Can you include an example implementation of extractHttpResponse?

File: README.markdown [hunk 3]
```diff
@@ -1,3 +1,76 @@
...
+You have to construct notification payload using one of the case classes from `com.gu.mobile.notifications.client.models.NotificationPayload` hierarchy
+and send that object through wire using `ApiClient.send` method. 
+#### Example: sending Breaking News alert
+```scala
+val payload = BreakingNewsPayload(
+  title = "Xi Jinping and Obama speak at Paris climate talks",
+  `type` = BreakingNews.toString,
+  message = "195 countries and nearly 150 world leaders including Barack Obama and Xi Jinping meet in Paris for COP21 UN climate change conference",
+  sender = "test sender",
+  editions = Set("UK", "US"),
```
Comment (davidfurey) [inline README.markdown]: I know this isn't related to this PR, but do you think editions and `type` should have a less generic type than String?

File: README.markdown [hunk 4]
```diff
@@ -1,3 +1,76 @@
...
+  imageUrl = None,
+  thumbnailUrl = None,
+  link = ExternalLink("http://mylink"),
+  importance = Importance.Major,
+  topic = Set(Topic("breaking", "environment/climate-change"))
```
Comment (davidfurey) [inline README.markdown]: Where does the magic string "breaking" come from?  Should it be here: https://github.com/guardian/mobile-notifications-api-client/blob/cd47ba5d4d116da1a33750692534ae632fc28895/src/main/scala/com/gu/mobile.notifications.client/models/legacy/Notification.scala#L9

**Comment by agulowaty** [comment]
@davidfurey: please put all comments regarding implementation here: https://github.com/guardian/mobile-notifications-api-client/pull/10

File: README.markdown (hunk 3 above)
Comment (agulowaty) [inline README.markdown]: fully agree

**Pull request closed without merging**